src/commonMain/kotlin/org/kbson/BsonType.kt

```diff
@@ -20,7 +20,7 @@ package org.kbson
  *
  * @property value the int value of this BSON type.
  */
-public enum class BsonType(public val value: Int) {
+public enum class BsonType(public val value: Byte) {
```

Comment:
I think MinKey breaks it - 0xFF.toByte() = -1

Reply:
Not sure what you mean.  0xFF is 255 when it's an unsigned byte, which I think is what we want.  Or leave it an int and then we don't have to worry about it going negative for MinKey.

Reply:
The reason I chose BsonType(value: Byte) is because when reading from bson or writing to bson its needed as a byte value. So I saved having to cast from int to byte by just using Byte. I have changed to UByte but now I have to convert to byte throughout the code.